src/efimon/perf/record.cpp

```diff
@@ -0,0 +1,128 @@
+/**
+ * @file record.cpp
+ * @author Luis G. Leon-Vega (luis.leon@ieee.org)
+ * @brief Observer to query the perf record command. It works as a wrapper
+ * for other functions like perf annotate
+ *
+ * @copyright Copyright (c) 2024. See License for Licensing
+ */
+
+#include <cstdlib>
+#include <efimon/perf/record.hpp>
+#include <filesystem>
+#include <iostream>
+#include <string>
+
+namespace efimon {
+
+extern uint64_t GetUptime();
+
+PerfRecordObserver::PerfRecordObserver(const uint pid,
+                                       const ObserverScope /* scope */,
+                                       const uint64_t interval,
+                                       const uint64_t frequency,
+                                       const bool no_dispose)
+    : Observer{}, valid_{false}, frequency_{frequency} {
+  this->pid_ = pid;
+  this->interval_ = interval;
+  this->no_dispose_ = no_dispose;
+  if (pid == 0) return;
+  if (interval == 0) interval_ = 1000;
+  if (frequency == 0) frequency_ = 1000;
+
+  this->CreateTemporaryFolder();
+  this->MakePerfCommand();
+}
```

Comment:
We need to configure this as a singleton. There could not be two instances with the same PID querying...